### PR TITLE
Add hawkular datawarehouse provider and example playbook

### PR DESCRIPTION
**Pull request by zgalor**
Added hawkular datawarehouse provider to provider ansible module.
Also added a playbook example.

@dkorn PTAL

### CODE REVIEW

File: library/manageiq_provider.py [hunk 1]
```diff
@@ -470,6 +472,8 @@ def main():
                 endpoints.append(manageiq.generate_openshift_endpoint('hawkular', 'hawkular', h_hostname, h_port, token))
         elif provider_type == "amazon":
             endpoints = [manageiq.generate_amazon_endpoint('default', 'default', access_key_id, secret_access_key)]
+        elif provider_type == "hawkular-datawarehouse":
+            endpoints = [manageiq.generate_openshift_endpoint('default', 'default', hostname, port, token)]
```
Comment (dkorn) [inline library/manageiq_provider.py]: we should probably rename `generate_openshift_endpoint` to `generate_auth_key_endpoint` and `generate_amazon_endpoint` to `generate_user_password_endpoint`.
@cben what do you think?

**Comment by dkorn** [comment]
@zgalor thanks for the contribution!

Please add the relevant details to the docs:

- manageiq_provider section in the README: https://github.com/dkorn/manageiq-ansible-module/blob/master/README.md#manageiq_provider-module

- provider types: https://github.com/dkorn/manageiq-ansible-module/pull/53/files#diff-1c9d8e7f51b7d3ee06e9a412ac9d0c79R48

- add an example: https://github.com/dkorn/manageiq-ansible-module/pull/53/files#diff-1c9d8e7f51b7d3ee06e9a412ac9d0c79R96

Also, doesn't have to be now, but please add tests for this provider :)

**Review by dkorn — CHANGES_REQUESTED** [review]

File: library/manageiq_provider.py [hunk 2]
```diff
@@ -155,7 +155,8 @@ class ManageIQ(object):
     PROVIDER_TYPES = {
         'openshift-origin': 'ManageIQ::Providers::Openshift::ContainerManager',
         'openshift-enterprise': 'ManageIQ::Providers::OpenshiftEnterprise::ContainerManager',
-        'amazon': 'ManageIQ::Providers::Amazon::CloudManager'}
+        'amazon': 'ManageIQ::Providers::Amazon::CloudManager',
+        'hawkular-datawarehouse': "ManageIQ::Providers::Hawkular::DatawarehouseManager"}
```
Comment (cben) [inline library/manageiq_provider.py]: Please drop `}` to next line, and add trailing comma.  Prettier [subjective] and will reduce future diffs.

File: library/manageiq_provider.py [hunk 3]
```diff
@@ -412,7 +413,7 @@ def main():
             name=dict(required=True),
             zone=dict(required=False, type='str'),
             provider_type=dict(required=True,
-                               choices=['openshift-origin', 'openshift-enterprise', 'amazon']),
+                               choices=['openshift-origin', 'openshift-enterprise', 'amazon', 'hawkular-datawarehouse']),
```
Comment (cben) [inline library/manageiq_provider.py]: `choices=PROVIDER_TYPES.keys()`

File: library/manageiq_provider.py (hunk 1 above)
Comment (cben) [reply library/manageiq_provider.py]: They actually generate {'endpoint':..., 'authentication': ...} pairs, so "endpoint" is not best term.
The right term is "connection configuration", TLDR...  How about shorter:
`generate_auth_key_config`
`generate_user_password_config`

However, I'd maybe leave the amazon one called `generate_amazon_config`.  It's endpoint part is rather degenerate, just role without host/port or anything like that, not sure it'll suit providers.
Perhaps it's cleanest o split endpoint from auth, but let's wait until we add more providers.

**Review by cben — COMMENTED** [review]
Nice.  Smaller changes than I expected.

Should this use `hawkular_hostname`, `hawkular_port` params instead of `hostname`, `port`?  Probably not.

**Comment by zgalor** [comment]
> Should this use hawkular_hostname, hawkular_port params instead of hostname, port? Probably not.

@cben we shouldn't. A dw provider will later apply to ElasticSearch. We should use the generic hostname and port.

**Comment by zgalor** [comment]
@dkorn @cben I made the changes you requested and added tests. Please review.

**Comment by cben** [comment]
LGTM

File: README.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ To try the modules copy and edit the relevant example playbook and execute:
 ### manageiq_provider module
 
 The `manageiq_provider` module currently supports adding, updating and deleting OpenShift and Amazon EC2 providers to manageiq.
```
Comment (dkorn) [inline README.md]: `supports adding, updating and deleting OpenShift, Amazon EC2 and **Hawkular** providers to manageiq`

File: tests/test_manageiq_provider.py [hunk 5]
```diff
@@ -157,24 +188,31 @@ def miq_ansible_module():
...
-        miq.generate_amazon_endpoint("default", "default", userid=AMAZON_USERID,
+        miq.generate_amazon_config("default", "default", userid=AMAZON_USERID,
                                      password=AMAZON_PASSWORD)]
 
 
+@pytest.fixture
+def hawk_dw_endpoint(miq):
+    yield [
+        miq.generate_auth_key_config("default", "default", HAWK_DW_HOSTNAME,
+                                        HAWK_DW_PROVIDER_PORT, HAWK_DW_PROVIDER_TOKEN)]
```
Comment (dkorn) [inline tests/test_manageiq_provider.py]: small nit, fix indent

File: tests/test_manageiq_provider.py [hunk 6]
```diff
@@ -157,24 +188,31 @@ def miq_ansible_module():
...
+        miq.generate_auth_key_config("hawkular", "hawkular", HAWKULAR_HOSTNAME,
                                         HAWKULAR_PORT, PROVIDER_TOKEN)]
 
 
 @pytest.fixture
 def amazon_endpoint(miq):
     yield [
-        miq.generate_amazon_endpoint("default", "default", userid=AMAZON_USERID,
+        miq.generate_amazon_config("default", "default", userid=AMAZON_USERID,
                                      password=AMAZON_PASSWORD)]
```
Comment (dkorn) [inline tests/test_manageiq_provider.py]: small nit, fix indent

File: tests/test_manageiq_provider.py [hunk 7]
```diff
@@ -157,24 +188,31 @@ def miq_ansible_module():
...
+        miq.generate_auth_key_config("default", "bearer", PROVIDER_HOSTNAME,
                                         PROVIDER_PORT, PROVIDER_TOKEN)]
 
 
 @pytest.fixture
 def hawkular_endpoint(miq):
     yield [
-        miq.generate_openshift_endpoint("hawkular", "hawkular", HAWKULAR_HOSTNAME,
+        miq.generate_auth_key_config("hawkular", "hawkular", HAWKULAR_HOSTNAME,
                                         HAWKULAR_PORT, PROVIDER_TOKEN)]
```
Comment (dkorn) [inline tests/test_manageiq_provider.py]: small nit, fix indent

File: tests/test_manageiq_provider.py [hunk 8]
```diff
@@ -197,8 +235,8 @@ def fail(msg):
     yield miq
 
 
-def test_generate_openshift_endpoint(miq):
-    endpoint = miq.generate_openshift_endpoint("default", "bearer",
+def test_generate_auth_key_config(miq):
+    endpoint = miq.generate_auth_key_config("default", "bearer",
                                                PROVIDER_HOSTNAME,
```
Comment (dkorn) [inline tests/test_manageiq_provider.py]: small nit, fix indent

File: library/manageiq_provider.py [hunk 9]
```diff
@@ -465,11 +481,13 @@ def main():
...
-            endpoints = [manageiq.generate_openshift_endpoint('default', 'bearer', hostname, port, token)]
+            endpoints = [manageiq.generate_auth_key_config('default', 'bearer', hostname, port, token)]
             if module.params['metrics']:
-                endpoints.append(manageiq.generate_openshift_endpoint('hawkular', 'hawkular', h_hostname, h_port, token))
+                endpoints.append(manageiq.generate_auth_key_config('hawkular', 'hawkular', h_hostname, h_port, token))
         elif provider_type == "amazon":
-            endpoints = [manageiq.generate_amazon_endpoint('default', 'default', access_key_id, secret_access_key)]
+            endpoints = [manageiq.generate_amazon_config('default', 'default', access_key_id, secret_access_key)]
+        elif provider_type == "hawkular-datawarehouse":
+            endpoints = [manageiq.generate_auth_key_config('default', 'default', hostname, port, token)]
```
Comment (dkorn) [inline library/manageiq_provider.py:490]: I think we should include `"hawkular-datawarehouse"` in the first providers `if` block `("openshift-enterprise", "openshift-origin")` and add `PROVIDER_AUTH_TYPES` dict in `ManageIQ` class, to include these constants `('default'` for hawkular, `'bearer'` for the openshift providers). 
Perhaps also do that for the roles.
@cben what is your take here?

Not a blocker for this change, can also be part of a future refactor

**Review by dkorn — CHANGES_REQUESTED** [review]

File: library/manageiq_provider.py (hunk 9 above)
Comment (cben) [reply library/manageiq_provider.py:490]: I'd wait till we actually add more providers, we'll be smarter then.
Won't object if you/Zohar decide to refactor now.

**Comment by zgalor** [comment]
@dkorn I fixed indentations and README.
Agree with @cben. We'll be smarter with the refactoring once we have more providers.

**Review by dkorn — APPROVED** [review]

**Pull request merged**